### PR TITLE
Add option to set custom ClusterID Tag Prefix

### PR DESCRIPTION
**Pull request by mikkeloscar**
Currently the controller will set a tag with the format: `kubernetes.io/cluster/<id>=owned` on the stack and this will be propagated down to the subnets. The tag has two purposes:
1. Identify resources as belonging to a particular subnet
2. Find the egress stack belonging to a particular cluster in order to manage the right stack in case you have multiple clusters in one AWS account/region.

The downside of the tag is that having such a tag on the subnets means that Kubernetes (Service Type Load Balancer logic) will consider the NAT subnets when finding subnets for ELBs. We never want to use the NAT subnets for this purpose so it's desirable to avoid tagging the subnets at all to prevent this from happening no matter how all the other subnets in the account are tagged.

This PR introduces a way to solve the problem by allowing to customize the tag prefix (`kubernetes.io/cluster/`) part of the tag. E.g. in context of zalando we could set a custom tag prefix: `zalando.org/cluster/` which would still support purpose 2 and also purpose 1 just by a slightly different tag structure.

The default stays the same.

### CODE REVIEW

File: provider/aws/aws.go [hunk 1]
```diff
@@ -32,11 +32,11 @@ const (
 	tagDefaultAZKeyRouteTableID         = "AvailabilityZone"
 	tagDefaultTypeValueRouteTableID     = "dmz" // find route table by "Type" tag = "dmz"
 	egressConfigTagPrefix               = "egress-config/"
-	clusterIDTagPrefix                  = "kubernetes.io/cluster/"
-	kubernetesApplicationTagKey         = "kubernetes:application"
-	resourceLifecycleOwned              = "owned"
-	maxStackWaitTimeout                 = 15 * time.Minute
-	stackStatusCheckInterval            = 15 * time.Second
+	// clusterIDTagPrefix                  = "kubernetes.io/cluster/"
```
Comment (szuecs) [inline provider/aws/aws.go:35]: line should be deleted

**Comment by szuecs** [comment]
:+1:

File: provider/aws/aws.go (hunk 1 above)
Comment (mikkeloscar) [reply provider/aws/aws.go]: removed!

**Comment by mikkeloscar** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**